base/c.jl

```diff
@@ -43,6 +43,35 @@ end
 
 typealias Coff_t FileOffset
 
+# C NUL-terminated string pointers; these can be used in ccall
+# instead of Ptr{Cchar} and Ptr{Cwchar_t}, respectively, to enforce
+# a check for embedded NUL chars in the string (to avoid silent truncation).
+const sizeof_Int = box(UInt32, unbox(Int32, Int.size))
+if sizeof_Int === 0x00000008 # need === since == doesn't exist yet
```

Comment:
wow, this is low level bit twiddling. the way boot.jl does this is probably cleaner (`if is(Int,Int64) ... end`)